frontend/src/components/EventFilter.js

```diff
@@ -0,0 +1,174 @@
+import React, { Component } from 'react';
+import { changeMapState } from "../actions/index";
+import { connect } from "react-redux";
+import { Button, Form } from 'react-bootstrap';
+
+// initialize global constant values
+const NONE_SELECTED_KEY = "";
+const SOCIAL_GATHERING_KEY = "0";
+const VOLUNTEER_EVENT_KEY = "1";
+const STUDENT_ORGANIZATION_EVENT_KEY = "2";
+
+const SOCIAL_GATHERING = "Social Gathering";
+const VOLUNTEER_EVENT = "Volunteer Event";
+const STUDENT_ORGANIZATION_EVENT = "Student Organization Event";
+const NO_FILTER_SELECTED = null;
+
+function mapDispatchToProps(dispatch) {
+  return {
+    changeMapState: mapState => dispatch(changeMapState(mapState))
+  };
+}
+
+const mapStateToProps = state => {
+  return { articles: state.articles };
+}
+
+class EventFilter extends Component {
+  constructor(props) {
+    super(props);
+    this.reduxState = this.props.articles[0];
+
+    if (this.reduxState) {
+      this.state = {
+        location: this.reduxState.location,
+        lat: this.reduxState.lat,
+        lng: this.reduxState.lng,
+        loggedIn: this.reduxState.loggedIn,
+        plusCode: this.reduxState.plusCode,
+        credentials: this.reduxState.credentials
+      };
+    }
+
+    // initialize global values that can change
+    this.isChecked = false;
+
+    this.handleFilterChange = this.handleFilterChange.bind(this);
+    this.updateReduxState = this.updateReduxState.bind(this);
+    this.handleDateChange = this.handleDateChange.bind(this);
+    this.handleFilterClearChange = this.handleFilterClearChange.bind(this);
+  }
+
+  async handleFilterClearChange() {
+    var category = this.reduxState.filter_choice;
+    var date = this.date;
+    this.updateReduxState(this.isChecked, category, date);
+
+    // const newState = {
```

Comment:
All this commented code unnecessary? Might as well get rid of it (and the function below)